src/domain/service/noteSettings.ts

```diff
@@ -131,7 +135,19 @@ export default class NoteSettingsService {
    * @returns team members
    */
   public async getTeamByNoteId(noteId: NoteInternalId): Promise<Team> {
-    return await this.teamRepository.getByNoteId(noteId);
+    let team = await this.teamRepository.getTeamByNoteId(noteId);
+    let parentId = await this.shared.note.getParentNoteIdByNoteId(noteId);
+
+    /**
+     * team.length === 1 means that team contains only creator or owner, it means that team is not spesified by user
+     * parentId === null means that note has no parent to inherit its team
+     */
+    while (team.length === 1 && parentId !== null) {
```

Comment:
can't figure out why we need `while`
Isn't `if` enough?

Reply:
than we will check only one parent (if it's team is not specified that we should check parent of the parent)